### ISSUE TITLE
Elias: Workflows, check in, Week 2

### ISSUE THREAD

**Issue by EliasMlopez99** [lab-brussels-1/home#135]
### 🥚 **[0. Local Development Without Git]**
Practice the foundational workflows of software development by learning to write Markdown locally on your own computer using Visual Studio Code (VSCode), the Command Line Interface (CLI), and NPM scripts to automate your code's quality (formatting, linting and spell checking).

 

- [ ] **_🥚 Folder Structures:_** You can explain how files and folders are stored in your computer and can find, open or create files in your computer without using the Command Line Interface.

_**🥚 Command Line Interface (CLI):**_ In a Unix shell you can ...

- [ ] -  open a new terminal window
- [ ] -  navigate up and down directories using cd
- [ ] -  list the contents of a directory using ls
- [ ] -  view the contents of a file using cat
- [ ] -  create new files using touch
- [ ] -  create new folders using mkdir

- [ ]  **_🥚 Markdown:_** You can write a document in Markdown with no syntax mistakes that renders into a well-formatted document.

🥚 _**VSCode:**_ You can complete these workflows in VScode, and can use keyboard shortcuts to complete them all:

- [ ]  Opening a repository in a new window
- [ ]  Opening the VSCode terminal
- [ ]  Adding a new file
- [ ]  Adding a new folder
- [ ]  Deleting a file
- [ ]  Deleting a folder
- [ ]  Previewing a Markdown File
- [ ]  Formatting a Markdown document

 

- [ ] 🥚 **_READMEs:_** You can write a README file that describes the project you are working on, why it's helpful, and how someone can use it.

🥚 **_NPM:_** You can use NPM commands to verify your code's quality, this includes ...
 
- [ ] -  using npm install to install a project's dependencies
- [ ] -  reading a package.json file to find which scripts are available for the project
- [ ] - use npm run <script> to execute an npm script

🥚 **_Formatting Code:_** You can use Prettier to make sure all the code in your project is well-formatted:
 

- [ ] You can use the Prettier VSCode extension to format a document while you are writing it
- [ ]  You can use npm run format to format all of the documents in your project
- [ ]  You can use npm run format:check to make sure all files are well-formatted

🥚 **_Linting Folder and File Names:_** You can ...

- [ ]  use npm run lint:ls to check all folder and file names in your project
- [ ]  you can fix all linting mistakes reported by npm run lint:ls

🥚 **_Linting Code:_** You can ...
 

- [ ] use npm run lint:md to check all Markdown files in your folder for linting mistakes
- [ ]  you can fix all linting mistakes reported by npm run lint:md

🥚 **_Spell Check:_** You can ...

- [ ]  use the Code Spell Checker VSCode extension to correct spelling mistakes in your Markdown documents while you are writing
- [ ]  use npm run spell-check to check the spelling in all the files of your project
- [ ]  update .cspell.json to add words that should be allowed in your project

🥚 **_File and Folder Naming Conventions:_** You can ...
 

- [ ] identify and follow the naming conventions for the project you are working on.
- [ ]  use npm run lint:ls to check that all files and folders follow the project's naming conventions.
- [ ]  you can correct any file and folder names to make them match the project conventions.

 

- [ ] 🐣 **_File Extensions:_** You can identify all of the languages covered at HYF and give the correct file extension. You don't need to know the languages, just recognize them.

 

- [ ] 🐣 **_Touch Typing:_** You can write a Markdown file without looking at your keyboard to find any letters, numbers or special characters. (slowly is ok!)

### **🥚 [1. Local Development With Git]**

Practice using Git to save and organize your development process. You will learn how you can use Git to go back to previous versions of your project, and to work on different changes in parallel.

🥚 **_Git: Using the CLI you can ..._**

- [ ]  Initialize a new git repository with git init
- [ ]  stage changes using git add <path>
- [ ]  check what is staged with git status
- [ ]  commit changes using git commit -m <message>
- [ ]  display your repository's git history using git log
- [ ]  create a new branch using git branch <branch-name>
- [ ]  check out a branch using git checkout <branch-name>
- [ ]  create a new branch and check it out using git checkout -b <branch-name>
- [ ]  merge changes from one branch to another using git merge <branch-name>
- [ ]  return to a previous version of your project with git log and git checkout <commit-hash>
- [ ]  stash and retrieve uncommitted changes with git stash and git pop

- [ ] 🥚 **_.gitignore:_** You can use a .gitignore file to describe which files you don't want included in your git history.

🥚 V**_SCode:_** You can ...
 

- [ ] use the Git Graph extension to to visualize your repo's commit history
- [ ]  use the Git Lens extension to investigate your repository's commit history
- [ ]  explain how the file tree in VSCode can show you which files have uncommitted changes

 

- [ ] 🥚 **_Atomic Commits:_** You can save your development progress using small commits with clear and helpful message.

 

- [ ] 🐣 **_Feature Branches:_** You can organize your development process using branches. You can create a new branch for each part of your project and merge those changes to main when they are finished.

**Comment by EliasMlopez99** [comment]
Week 1 Workflows


- [ ] I have pushed my progress to [my fork of exercises repo](_)

## I Need Help With
I need to practice the git pulling and git pushing process, I know that this process requires a well sequence of commands, that if not well performed, can cause several problems 
## What went well?

- Creating a new branch from my local machine with the help of the terminal from GitHub
- Git cloning a public repository
- Git pulling and git pushing 
- Pulling a request 
- committing my changes in my branch
- Understood concept of Staging 

## What went less well?

- At the beginning I had problems with creating new branches
- I have problems to merge branches
- how to properly delete unused files, I accidentally erase my main branch in the effort to erase another branch that was not being used

## Lessons Learned
I understood how to git clone a public repo to my local machine and and how to create a new branch from my local directory to pull a request in a certain git hub repository, in this case it was the hack your future repo, I also learned how to git push and git pull and push commands needed to work in your local machine. basically so far what I have learned is that Github is a tool that let programmers to construct a whole project where other team members can contribute before bringing this project to reality, It is like the construction plans for a house, everyone contributes to build an idea that later on will be created.

## Sunday Prep Work
I still have stuff to complete in my issues list that I hope to have it ready by Saturday
starting from NPM all below

**Comment by danielhalasz** [comment]
@EliasMlopez99 nice to see that you are experimenting and practicing with git, that is the best way to fully understand it.

at this point, you do not really need to do git merge locally, so it is fine if it is not fully clear now.

to delete a branch, first you would switch to another branch (like main) and then run `git branch -D branchname`

**Comment by danielhalasz** [comment]
@EliasMlopez99 week-2?

**Comment by EliasMlopez99** [comment]
### Week 2

- [ ]  I have pushed my progress to [my fork of exercises repo](https://github.com/lab-brussels-1/home/issues/_)

### I Need Help With:

So far everything is clear with the exception of how to control the size of an image when you use the following code
`![alt text for your image](/path/to/image.png)` I would like to know how do you do it.

### What went well?

Pulling a request, git pulling and git pushing, and running the npm commands in order to check my codification

### What went less well?

the `cspell.json` it has been a problem since the beginning, though I could upload my file because I ran the `npm` commands, I needed first add words to this section to avoid conflicts with my files.

### Lessons Learned

1. I know where I am know, I have been practicing a lot by myself and did a lot of mistakes but now is more clear, practicing helps more than reading.
2. I fully understand how `npm` works and how to run the commands in a logic sequence.
3. I now know that in order to avoid conflicts with the spell check I have to update our spell.json

### Sunday Prep Work
Preparing myself studying and watching material related to HTML codification.

**Comment by danielhalasz** [comment]
@EliasMlopez99 indeed, it is not possible to change the image since in markdown (unless using HTML and inline CSS, which break the linting check). What you can do is to edit the file itself and then add the resized version. Cspell.json is designed to allow new words to be added, in order to pass the spell check. So it is all normal that you are experiencing now 😄

**Comment by minjupgeorge** [comment]
hi @danielhalasz , @EliasMlopez99 ,

![Kochi](https://images.unsplash.com/photo-1590123732197-e7079d2ceb89?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8MXx8a29jaGl8ZW58MHx8MHx8&w=1000&q=80impolicy=website&width=850&height=300)
I Have use the above format for website link and its working . Please correct me if am wrong

**Comment by danielhalasz** [comment]
it might work in the case of unsplash, because the URL is formatted in a way that the server serves a small image, but this is very specific to Unsplash and does not work with images that you put in a folder in your repo

**Comment by minjupgeorge** [comment]
> 

@danielhalasz  ok I understood. Thanks for correcting me.
